plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3326,4 +3327,51 @@ public void testOptimizeForPartitionedTable()
         assertThat(updatedFiles).hasSize(3);
         assertThat(getAllDataFilesFromTableDirectory(tableName)).containsExactlyInAnyOrderElementsOf(concat(initialFiles, updatedFiles));
     }
+
+    @Test
+    public void testVacuumParameterValidation()
+    {
+        assertQueryFails(
+                "ALTER TABLE no_such_table_exists EXECUTE VACUUM",
+                "\\Qline 1:1: Table 'iceberg.tpch.no_such_table_exists' does not exist");
+        assertQueryFails(
+                "ALTER TABLE nation EXECUTE VACUUM (retention_threshold => '33')",
+                "\\QUnable to set procedure property 'retention_threshold' to ['33']: duration is not a valid data duration string: 33");
+        assertQueryFails(
+                "ALTER TABLE nation EXECUTE VACUUM (retention_threshold => '33mb')",
+                "\\QUnable to set procedure property 'retention_threshold' to ['33mb']: Unknown time unit: mb");
+    }
+
+    @Test
+    public void testVacuum()
+            throws InterruptedException
```

Comment:
We normally do `throws Exception` for test methods that throw any checked exceptions